datafusion/sql/src/expr/identifier.rs

```diff
@@ -47,40 +47,58 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             // compound identifiers, but this is not a compound
             // identifier. (e.g. it is "foo.bar" not foo.bar)
             let normalize_ident = self.normalizer.normalize(id);
-            match schema.field_with_unqualified_name(normalize_ident.as_str()) {
-                Ok(_) => {
-                    // found a match without a qualified name, this is a inner table column
-                    Ok(Expr::Column(Column {
-                        relation: None,
+
+            // Check for qualified field with unqualified name
+            if let Ok((Some(qualifier), _)) =
+                schema.qualified_field_with_unqualified_name(normalize_ident.as_str())
```

Comment:
I think it can be simplified to
```rust
            if let Ok((qualifier, _)) =
                schema.qualified_field_with_unqualified_name(normalize_ident.as_str())
            {
                return Ok(Expr::Column(Column {
                    relation: qualifier.filter(|q| q.table() != UNNAMED_TABLE).cloned(),
                    name: normalize_ident,
                }));
            }
```
And we don't need to check again for unqualified field.  
`qualified_field_with_unqualified_name()` also returns unqualified fields.

Reply:
sure